sdk/core/azure-core/inc/azure/core/paged_response.hpp

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Provides support for paged responses.
+ */
+
+#pragma once
+
+#include <cstdlib>
+#include <string>
+
+#include "azure/core/context.hpp"
+#include "azure/core/http/raw_response.hpp"
+#include "azure/core/nullable.hpp"
+
+namespace Azure { namespace Core {
+
+  /**
+   * @brief Defines the base type and behavior for a paged response.
+   *
+   * @remark The template is using for making static-inheritance.
+   *
+   * @remark T classes must implement the way to get and move to the next page.
+   *
+   * @tparam T A class type for static-inheritance.
+   */
+  template <class T> class PagedResponse {
```

Comment:
nit: Keep this as `Derived` and update comments.

```suggestion
  template <class Derived> class PagedResponse {
```

Reply:
I like using `T` for type in templates.  Its very idiomatic to the language.

Reply:
From @CaseyCarter :
> it's fairly idiomatic to use name the template argument `Derived` or some variation thereof when implementing CRTP. The singular example in the C++ Standard Library is `template <class D> class view_interface;` ([view.interface.general](http://eel.is/c++draft/view.interface.general)).

See previous discussion on this:
https://github.com/Azure/azure-sdk-for-cpp/pull/2116#discussion_r613201979

Reply:
FWIW, renaming the template argument name isn't a breaking change, and I don't feel strongly about this. Rick, we can leave it as is, if you prefer it.

Reply:
@ahsonkhan  https://github.com/Azure/azure-sdk-for-cpp/pull/2116/files#r613201979

I knew I read this comment before 🍡